src/components/interactive-builder/modals/question/question-form/question/question.test.tsx

```diff
@@ -1,73 +1,182 @@
 import React from 'react';
 import { render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
-import type { FormField } from '@openmrs/esm-form-engine-lib';
-import { FormFieldProvider } from '../../form-field-context';
 import Question from './question.component';
+import { FormFieldProvider } from '../../form-field-context';
+import type { FormField } from '@openmrs/esm-form-engine-lib';
 
-const mockSetFormField = jest.fn();
-const checkIfQuestionIdExists = jest.fn();
+jest.mock('react-i18next', () => ({
+  useTranslation: jest.fn(() => ({
+    t: (key, fallback) => fallback || key,
+  })),
+}));
 
-const formField: FormField = {
+const initialFormField: FormField = {
+  id: 'testId',
+  label: 'Test Label',
   type: 'obs',
-  questionOptions: { rendering: 'text' },
-  id: '1',
+  questionOptions: {
+    rendering: 'text',
+  },
 };
 
-jest.mock('../../form-field-context', () => ({
-  ...jest.requireActual('../../form-field-context'),
-  useFormField: () => ({ formField, setFormField: mockSetFormField }),
-}));
+const renderWithFormFieldProvider = (
+  ui: React.ReactElement,
+  { formField = initialFormField, selectedConcept = null } = {},
+) => {
+  return render(
+    <FormFieldProvider initialFormField={formField} selectedConcept={selectedConcept}>
+      {ui}
+    </FormFieldProvider>,
+  );
+};
 
 describe('Question Component', () => {
-  it('should render the toggles for question info', () => {
-    renderQuestionComponent();
-
-    expect(
-      screen.getByRole('group', { name: /would you like to provide additional details about the question?/i }),
-    ).toBeInTheDocument();
-    expect(screen.getByRole('radio', { name: /Yes/i })).toBeInTheDocument();
-    expect(screen.getByRole('radio', { name: /No/i })).toBeInTheDocument();
+  const checkIfQuestionIdExists = jest.fn(() => false);
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should render all required fields', () => {
+    renderWithFormFieldProvider(<Question checkIfQuestionIdExists={checkIfQuestionIdExists} />);
+
+    expect(screen.getByRole('textbox', { name: /Question ID/i })).toBeInTheDocument();
```

Comment:
Since the `/i` is there, the text doesn't need to strictly be in the case we expect.
```suggestion
    expect(screen.getByRole('textbox', { name: /question id/i })).toBeInTheDocument();
```